drivers/resetvdis.py

```diff
@@ -53,10 +53,23 @@ def reset_sr(session, host_uuid, sr_uuid, is_sr_master):
     sr_lock.release()
     gc_lock.release()
 
-def reset_vdi(session, vdi_uuid, force, term_output=True, writable=True):
-    vdi_ref = session.xenapi.VDI.get_by_uuid(vdi_uuid)
-    vdi_rec = session.xenapi.VDI.get_record(vdi_ref)
-    sm_config = vdi_rec["sm_config"]
+def reset_vdi(session, vdi_uuid, force, term_output=True, writable=True, vdiref=None, srref=None, smconf={}):
```

Comment:
You can call vdiref -> vdi_ref, srref -> sr_ref, smconf -> sm_conf for better readability and to improve implementation. See below